### PR TITLE
Vep update

### PR DESCRIPTION
**Pull request by ramprasadn**
## PR checklist

Addresses https://github.com/nf-core/raredisease/issues/477 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/raredisease/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/raredisease _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test_one_sample,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: conf/test.config [hunk 1]
```diff
@@ -58,10 +58,5 @@ params {
     vep_cache            = "https://raw.githubusercontent.com/nf-core/test-datasets/raredisease/reference/vep_cache_and_plugins.tar.gz"
     vep_filters          = "https://raw.githubusercontent.com/nf-core/test-datasets/raredisease/reference/hgnc.txt"
     vep_cache_version    = 107
```
Comment (jemten) [inline conf/test.config:60]: needed or update to version 110?

File: conf/test_one_sample.config [hunk 2]
```diff
@@ -58,10 +58,5 @@ params {
     vep_cache            = "https://raw.githubusercontent.com/nf-core/test-datasets/raredisease/reference/vep_cache_and_plugins.tar.gz"
     vep_filters          = "https://raw.githubusercontent.com/nf-core/test-datasets/raredisease/reference/hgnc.txt"
     vep_cache_version    = 107
```
Comment (jemten) [inline conf/test_one_sample.config:60]: 110?

File: nextflow_schema.json [hunk 3]
```diff
@@ -383,6 +383,15 @@
                     "help_text": "If no directory path is passed, vcf files will not be annotated by vep.",
                     "fa_icon": "fas fa-folder-open"
                 },
+                "vep_plugin_files": {
```
Comment (jemten) [inline nextflow_schema.json:386]: could you add a specific schema for this input. Mainly checking that the files exists.

Comment (jemten) [reply nextflow_schema.json:386]: also, don't you need to add this file to option to `nextflow.config`?

File: nextflow_schema.json [hunk 4]
```diff
@@ -383,6 +383,15 @@
...
+                    "type": "string",
+                    "exists": true,
+                    "format": "file-path",
+                    "description": "Databases used by both named and custom plugins to annotate variants.",
+                    "fa_icon": "fas fa-file-csv",
+                    "help_text": "Path to a file containing the absolute paths to databases and their indices used by VEP's custom and named plugins resources defined within the vcfanno toml file. One line per resource.",
+                    "mimetype": "text/csv"
```
Comment (jemten) [inline nextflow_schema.json:393]: could you add a pattern arrument to ensure `.csv` file ending

File: subworkflows/local/annotate_genome_snvs.nf [hunk 5]
```diff
@@ -115,20 +116,23 @@ workflow ANNOTATE_GENOME_SNVS {
             }
             .set { ch_for_mix }
 
-        ch_vep_in = ch_for_mix.selvar.mix(ch_for_mix.cadd)
+        ch_for_mix.selvar.mix(ch_for_mix.cadd)
+            .map { meta, vcf -> return [meta, vcf, []]}
```
Comment (jemten) [inline subworkflows/local/annotate_genome_snvs.nf:120]:
```suggestion
            .map { meta, vcf -> return [meta, vcf, []] }
```

File: subworkflows/local/annotate_mobile_elements.nf [hunk 6]
```diff
@@ -54,18 +54,21 @@ workflow ANNOTATE_MOBILE_ELEMENTS {
             ch_genome_fasta,
             ch_genome_dictionary
         )
+        .vcf
+        .map { meta, vcf -> return [meta, vcf, []]}
```
Comment (jemten) [inline subworkflows/local/annotate_mobile_elements.nf:58]:
```suggestion
        .map { meta, vcf -> return [meta, vcf, []] }
```

File: subworkflows/local/annotate_mt_snvs.nf [hunk 7]
```diff
@@ -49,22 +51,27 @@ workflow ANNOTATE_MT_SNVS {
                     return [it[0], it[2]]
             }
             .set { ch_for_mix }
-        ch_vep_in = ch_for_mix.merged.mix(ch_for_mix.cadd)
+
+        ch_for_mix.merged.mix(ch_for_mix.cadd)
+            .tap { ch_haplogrep_in }
+            .map { meta, vcf -> return [meta, vcf, []]}
```
Comment (jemten) [inline subworkflows/local/annotate_mt_snvs.nf:57]:
```suggestion
            .map { meta, vcf -> return [meta, vcf, []] }
```

**Review by jemten — COMMENTED** [review]
Nice! Some minor comments.

File: conf/test.config (hunk 1 above)
Comment (ramprasadn) [reply conf/test.config:60]: Test data is from 107 that's why I haven't changed it here :)

**Pull request merged**